service/provider/src/main/java/gov/va/vro/service/provider/mas/service/MasApiService.java

```diff
@@ -0,0 +1,137 @@
+package gov.va.vro.service.provider.mas.service;
+
+import com.fasterxml.jackson.core.type.TypeReference;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import gov.va.vro.service.provider.MasApiProps;
+import gov.va.vro.service.provider.mas.MasException;
+import gov.va.vro.service.provider.mas.model.*;
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.http.*;
+import org.springframework.security.oauth2.core.OAuth2AccessToken;
+import org.springframework.stereotype.Component;
+import org.springframework.web.client.RestClientException;
+import org.springframework.web.client.RestTemplate;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+
+@Component
+@RequiredArgsConstructor
+@Slf4j
+public class MasApiService {
+  private final RestTemplate restTemplate;
+  private final MasAuthToken masAuthToken;
+  private final MasApiProps masApiProps;
+
+  public List<MasCollectionStatus> getMasCollectionStatus(List<Integer> collectionIds)
+      throws MasException {
+    try {
+      String url = masApiProps.getBaseURL() + masApiProps.getCollectionStatusPath();
+      HttpHeaders headers = getMasHttpHeaders();
+      List<MasCollectionStatusReq> masCollectionStatusReqList =
+          new ArrayList<MasCollectionStatusReq>(collectionIds.size());
```

Comment:
Do you need to collectionIds.size() and MasCollectionStatusReq here?

Reply:
fixed